src/main/java/reposense/git/GitClone.java

```diff
@@ -100,21 +100,39 @@ public static void cloneShallowPartialBare(RepoConfiguration config, Path rootPa
      *
      * @throws GitCloneException when an error occurs while attempting to clone the repo.
      */
-    public static void clone(RepoConfiguration config) throws GitCloneException {
+    public static void cloneAndBranch(RepoConfiguration config) throws GitCloneException {
+        String outputFolderName = Paths.get(FileUtil.getRepoParentFolder(config).toString(),
+                config.getRepoName()).toString();
+        clone(config, Paths.get("."), outputFolderName);
         try {
-            Path rootPath = FileUtil.getRepoParentFolder(config);
-            Path repoPath = Paths.get(rootPath.toString(), config.getRepoName());
+            config.updateBranch();
+            GitCheckout.checkout(config.getRepoRoot(), config.getBranch());
+        } catch (GitBranchException gbe) {
+            logger.log(Level.SEVERE,
+                    "Exception met while trying to get current branch of repo. Analysis terminated.", gbe);
+            throw new GitCloneException(gbe);
+        } catch (RuntimeException rte) {
+            logger.log(Level.SEVERE, "Branch does not exist! Analysis terminated.", rte);
```

Comment:
Is there a way to test either `catch` branches?

Reply:
I believe the 'branch does not exist' scenario is already covered in `systemTest` as the `repo-config.csv` that they use has a `nonExistentBranch` for its second last repo.

For the `exception met while trying to get current branch` case, I'm actually not too sure if we can force this exception to occur. It sounds like something that only happens when command runner outright fails.

Reply:
After doing a bit of digging, I remembered that you pointed out that this method isn't used in the main codebase. In that case, I think it actually makes more sense to get rid of all these logging messages and retain the barebones functionality of the method itself to simplify it. 

Branch checking is actually done with `GitRevParse` in the main code.

Reply:
Should we also move the `cloneBare` method to the new `TestRepoCloner` class?

Reply:
I agree! I didn't realise `cloneBare` was also entirely not used in the main code. All instances were replaced with `cloneBareAsync`. I also don't think we'll be changing any of the cloning settings in the main code any time soon, so I think it might be better to remove it for now.

I have checked the remainder methods and they seem to be used in the main code in some form, so I have left them untouched.